compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -3209,6 +3209,10 @@ private void defineFieldsOfClassDef(BLangClassDefinition classDefinition, Symbol
 
         for (BLangSimpleVariable field : classDefinition.fields) {
             defineNode(field, typeDefEnv);
+            if (field.expr != null) {
+                field.symbol.flags |= Flags.OPTIONAL;
+                field.symbol.isDefaultable = true;
```

Comment:
If going with the `isDefaultable` field, shouldn't there be BIR changes as well?

Reply:
Updated in 0366d7780ccb161e07863fb7e1bb5cae8d34c0ef

Reply:
Removed OPTIONAL flag in 90ce5e1e8bbf22dd5988c8285af2b9304c213674